lib/Slackbot_worker.js

```diff
@@ -86,10 +105,22 @@ module.exports = function(botkit, config) {
             bot.rtm = new Ws(res.url, null, {agent: agent});
             bot.msgcount = 1;
 
-            var pingIntervalId = null;
+            bot.rtm.on('pong', function (obj) {
+                lastPong = Date.now()
+                botkit.debug('PONG received');
+            })
+
             bot.rtm.on('open', function() {
 
                 pingIntervalId = setInterval(function() {
+                    if (lastPong && lastPong + 12000 < Date.now()) {
```

Comment:
Since `lastPong` is initialized to `0` (line 15, line 61), and through my testing it appears that Slack's websocket server does NOT respond to WS "ping" messages with a "pong" ... `lastPong` will always be `0` and this `if` condition will always evaluate to false.

Have you ever actually observed the error inside this `if` conditional occurring?